python_modules/dagster/dagster/_utils/yaml_utils.py

```diff
@@ -152,3 +159,98 @@ def dump_run_config_yaml(run_config: Mapping[str, Any], sort_keys: bool = True)
         allow_unicode=True,
         sort_keys=sort_keys,
     )
+
+
+def parse_yaml_with_source_positions(
+    src: str,
+    filename: str = "<string>",
+    implicits_to_remove: Sequence[str] = ["tag:yaml.org,2002:timestamp"],
+) -> ValueAndSourcePositionTreeNode:
+    """Parse YAML source with source position information.
+    This function takes a YAML source string and an optional filename, and returns a
+    `ValueAndSourcePositionTreeNode` object. The `ValueAndSourcePositionTreeNode` contains the
+    parsed YAML value and a `SourcePositionTreeNode` that holds the source position information for
+    the YAML structure.
+
+    The function uses a custom YAML loader (`SafeLineLoader`) that tracks the source position
+    information for each YAML node. It also allows for the removal of specific implicit resolvers,
+    such as the default timestamp resolver.
+
+    Args:
+        src (str): The YAML source string to be parsed.
+        filename (str): The filename associated with the YAML source, used for error reporting.
+            Defaults to "<string>" if not provided.
+        implicits_to_remove (list[str]): A list of YAML implicit resolvers to remove. Defaults to
+            removing the timestamp resolver.
+
+    Returns:
+        ValueAndSourcePositionTreeNode: An object containing the parsed YAML value and the corresponding source
+            position information.
+    """
+
+    class SafeLineLoader(yaml.SafeLoader):
+        @classmethod
+        def remove_implicit_resolver(cls, tag_to_remove: Any) -> None:
+            if "yaml_implicit_resolvers" not in cls.__dict__:
+                cls.yaml_implicit_resolvers = cls.yaml_implicit_resolvers.copy()
+
+            for first_letter, mappings in cls.yaml_implicit_resolvers.items():
+                cls.yaml_implicit_resolvers[first_letter] = [
+                    (tag, regexp) for tag, regexp in mappings if tag != tag_to_remove
+                ]
+
+        def construct_object(self, node: Any, deep: Any = False) -> ValueAndSourcePositionTreeNode:
+            value = super().construct_object(node, True)
+            source_position = SourcePosition(
+                filename=filename,
+                start=LineCol(
+                    line=node.start_mark.line + 1,
```

Comment:
assume this is zero-index-to-one-index? might be worth noting inline or in `LineCol` docstring that that is the case